python/src/trezorlib/ethereum.py

```diff
@@ -50,10 +59,14 @@ def typeof_array(type_name: str) -> str:
 
 def parse_type_n(type_name: str) -> int:
     """Parse N from type<N>. Example: "uint256" -> 256."""
-    return int(re.search(r"\d+$", type_name).group(0))
+    match = re.search(r"\d+$", type_name)
+    if match:
+        return int(match.group(0))
+    else:
+        raise ValueError(f"Could not parse type<N> from {type_name}.")
 
 
-def parse_array_n(type_name: str) -> Union[int, str]:
+def parse_array_n(type_name: str) -> Union[int, Literal["dynamic"]]:
```

Comment:
much easier: `-> Optional[int]`.
it matches the (only) usage below too

Reply:
Done in b70acaf, good idea